src/toxcore/dht/daemon_state.rs

```diff
@@ -148,7 +156,7 @@ mod tests {
         let serialized_len = serialized_vec.len();
         let res = DaemonState::deserialize_old(&alice, &serialized_vec[..serialized_len - 1]).wait();
         assert!(res.is_err());
-        assert_eq!(*res.err().unwrap().kind(), DeserializeErrorKind::IncompleteData { needed: Needed::Size(55), data: serialized_vec[..serialized_len - 1].to_vec() });
+        assert_eq!(*res.err().unwrap().kind(), DeserializeErrorKind::IncompleteData { needed: Needed::Size(39), data: serialized_vec[..serialized_len - 1].to_vec() });
```

Comment:
Why has it changed?

Reply:
Why?

Reply:
I changed take! --> bytes::complete::take.
And now `Eof` is correct.
https://github.com/Geal/nom/pull/937/files#diff-5fbfee103ff92c86f212b5ae921ee6f4R373